src/test/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImplTest.java

```diff
@@ -1001,15 +1005,108 @@ void createAcspMembersDataWithDefaultProfileValues() throws DataException {
         verify(acspMembersService).create(any(AcspMembersSpec.class));
     }
 
+    @Test
+    void createAcspMembersDataWhenProfileIsNotNull() throws DataException {
+        AcspMembersSpec spec = new AcspMembersSpec();
+        spec.setUserId("userId");
+
+        AcspProfileData acspProfileData;
+        acspProfileData = new AcspProfileData("acspNumber");
+        AcspMembersData acspMembersData;
+        acspMembersData = new AcspMembersData("memberId", "acspNumber", "userId", "active", "role");
+        var acspStatus = "active";
+        var acspType = "ltd";
+        var supervisoryBody = "financial-conduct-authority-fca";
+        var membershipDetails = "Membership ID: FCA654321";
+
+        AcspProfileSpec acspProfile = new AcspProfileSpec();
+        acspProfile.setStatus(acspStatus);
+        acspProfile.setType(acspType);
+        AmlSpec amlSpec = new AmlSpec();
+        amlSpec.setSupervisoryBody(supervisoryBody);
+        amlSpec.setMembershipDetails(membershipDetails);
+
+        acspProfile.setAmlDetails(Collections.singletonList(amlSpec));
+
+        spec.setAcspProfile(acspProfile);
+
+        when(acspProfileService.create(any(AcspProfileSpec.class))).thenReturn(acspProfileData);
+        when(acspMembersService.create(any(AcspMembersSpec.class))).thenReturn(acspMembersData);
+
+        AcspMembersData result = testDataService.createAcspMembersData(spec);
+
+        assertNotNull(result);
+        assertEquals("acspNumber", result.getAcspNumber());
+        assertEquals(acspStatus, spec.getAcspProfile().getStatus());
+        assertEquals(acspType, spec.getAcspProfile().getType());
+        assertEquals(supervisoryBody,
+                spec.getAcspProfile().getAmlDetails().getFirst().getSupervisoryBody());
+        assertEquals(membershipDetails,
+                spec.getAcspProfile().getAmlDetails().getFirst().getMembershipDetails());
+        assertEquals("memberId", result.getAcspMemberId());
+        assertEquals("userId", result.getUserId());
+        assertEquals("active", result.getStatus());
+        assertEquals("role", result.getUserRole());
+
+        verify(acspProfileService).create(argThat(profile ->
+                acspType.equals(profile.getType()) && acspStatus.equals(profile.getStatus())
+                        && profile.getAmlDetails() != null && !profile.getAmlDetails().isEmpty()
+                        && supervisoryBody.equals(
+                                profile.getAmlDetails().getFirst().getSupervisoryBody())
+                        && membershipDetails.equals(
+                                profile.getAmlDetails().getFirst().getMembershipDetails())
+        ));
+
+        verify(acspMembersService).create(argThat(membersSpec ->
+                acspMembersData.getUserId().equals(membersSpec.getUserId())
+        ));
+    }
+
+    @Test
+    void createAcspMembersDataWhenProfileIsNull() throws DataException {
```

Comment:
Isn't this the same as `createAcspMembersDataWithDefaultProfileValues` and `createAcspMembersData`

Can we please add the new assertions to those existing tests (they both look identical to me so maybe one could go)